src/garbage.erl

```diff
@@ -6,6 +6,7 @@ garbage_leaves(KeeperLeaves, CFG) ->
     delete_stuff(0, KL, ids:leaf(CFG)),
     delete_stuff(0, [0|KeeperStems], ids:stem(CFG)),
     ok.
+-spec garbage([stem:stem_p()], cfg:cfg()) -> ok.
```

Comment:
@zack-bitcoin I read your comment at https://github.com/lucafavatella/MerkleTrie/commit/8a9c3969d13002523e6b019e2de25a3302732541#commitcomment-23049551 that refers to the same line of this PR, so I am answering here.

> I don't know much about dialyzer, so maybe I am mistaken.
> It looks like KeeperRoots in garbage collection is being restricted to being a list of one element.
> It is supposed to be a list of arbitrary length.

@zack-bitcoin `[stem:stem_p()]` means list of arbitrary length (and potentially empty), where each element in the list is of type `stem:stem_p()`. I am still not sure whether it makes sense to perform garbage collection keeping nothing... so I was thinking of restricting that type to non-empty list i.e. `[stem:stem_p(), ...]`.

Cc @domhaobaobao

(The language used for expressing types is defined [here](http://erlang.org/doc/reference_manual/typespec.html). Type specification `[T]` is equivalent to `list(T)`.)